files/src/main/java/com/fincity/saas/files/service/FilesSecuredAccessService.java

```diff
@@ -0,0 +1,60 @@
+package com.fincity.saas.files.service;
+
+import java.time.LocalDateTime;
+
+import org.jooq.types.ULong;
+import org.springframework.stereotype.Service;
+
+import com.fincity.nocode.reactor.util.FlatMapUtil;
+import com.fincity.saas.commons.jooq.service.AbstractJOOQDataService;
+import com.fincity.saas.commons.jooq.util.ULongUtil;
+import com.fincity.saas.commons.util.BooleanUtil;
+import com.fincity.saas.files.dao.FilesSecuredAccessKeyDao;
+import com.fincity.saas.files.dto.FilesSecuredAccessKey;
+import com.fincity.saas.files.jooq.tables.records.FilesSecuredAccessKeyRecord;
+
+import reactor.core.publisher.Mono;
+
+@Service
+public class FilesSecuredAccessService extends
+        AbstractJOOQDataService<FilesSecuredAccessKeyRecord, ULong, FilesSecuredAccessKey, FilesSecuredAccessKeyDao> {
+
+	public Mono<FilesSecuredAccessKey> getAccessRecordByPath(String key) {
+		return this.dao.getAccessByKey(key);
+	}
+
+	public Mono<String> getAccessPathByKey(String accessKey) {
+
+		return FlatMapUtil.flatMapMono(
+
+		        () -> this.getAccessRecordByPath(accessKey),
+
+		        accessKeyObject -> this.checkAccessWithinTime(accessKeyObject)
+		                .flatMap(BooleanUtil::safeValueOfWithEmpty),
```

Comment:
Why do you need the SafeValueOfWithEmpty??? You will be returning either true or false in this case right.